src/zkevm_specs/evm/execution/log.py

```diff
@@ -0,0 +1,60 @@
+from ..instruction import Instruction, Transition
+from ..table import CallContextFieldTag, TxLogFieldTag, TxContextFieldTag
+from ..opcode import Opcode
+from ...util.param import LOG_STATIC_GAS
+
+
+def log(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+    # constrain op in [log0, log4] range
+    instruction.range_lookup(opcode - Opcode.LOG0, 5)
+
+    # pop `mstart`, `msize` from stack
+    mstart = instruction.rlc_to_int_exact(instruction.stack_pop(), 8)
+    msize = instruction.rlc_to_int_exact(instruction.stack_pop(), 8)
```

Comment:
this is dynamic size. So you need to use multiple steps to copy the memory, similar to `calldatacopy`.

Reply:
thanks , fixed in [93e57b2](https://github.com/appliedzkp/zkevm-specs/pull/125/commits/93e57b2ef33976b72f57bec7b0ed1b4fb43046f0)